### PR TITLE
Quick match rv

### PR DESCRIPTION
**Pull request by carolyn-wang**
* Added places/locations dummy values & created Place object in database
* Show location of hangout on each hangout card
* Added recyclerView to show all possible Quick Match Hangouts
* created floating button that creates new hangout -- currently leads to blank fragment
image

### CODE REVIEW

File: app/src/main/java/com/example/friendly/HangoutQuery.java [hunk 1]
```diff
@@ -8,24 +8,47 @@
...
+        query.include(Hangout.KEY_LOCATION);
         query.setLimit(POSTS_TO_LOAD);
-        query.addDescendingOrder(Hangout.KEY_CREATED_AT);
+        // display past hangouts
+        if (queryConditions.contains("past")){
+            query.whereLessThan(Hangout.KEY_DATE, new Date());
+            query.addDescendingOrder(Hangout.KEY_DATE);
+        }
+        // display future hangouts
+        if (queryConditions.contains("future")){
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/HangoutQuery.java:37]: should these be else if {} blocks or will the query apply to all cases that match?
```
if (queryConditions.contains("past")) {
} else if (queryConditions.contains("user")){
} else if (...
} else {}
```

File: app/src/main/java/com/example/friendly/NavigationUtils.java [hunk 2]
```diff
@@ -53,16 +58,24 @@ public static void displayFragmentQuickMatch(ParseUser user, FragmentManager fra
         ft.commit();
     }
 
+    public static void displayCreateQuickMatch(ParseUser user, FragmentManager fragmentManager) {
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+        Fragment createQuickMatchFragment = new CreateQuickMatchFragment();
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/NavigationUtils.java:63]: can inline this since you aren't using it again
```
ft.replace(R.id.flContainer, new CreateQuickMatchFragment());
```

File: app/src/main/java/com/example/friendly/activities/MainActivity.java [hunk 3]
```diff
@@ -20,6 +23,7 @@ public class MainActivity extends AppCompatActivity {
 
     private static final String TAG = "MainActivity";
     private Context mContext;
+    private Activity mActivity;
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/activities/MainActivity.java:26]: Why do you need to keep the activity?

File: app/src/main/java/com/example/friendly/NavigationUtils.java [hunk 4]
```diff
@@ -53,16 +58,24 @@ public static void displayFragmentQuickMatch(ParseUser user, FragmentManager fra
...
         ft.replace(R.id.flContainer, longMatchFragment);
         ft.commit();
     }
 
-    public static void displayFragmentHangoutHistory(ParseUser user, FragmentManager fragmentManager) {
+    public static void displayFragmentHangoutHistory(FragmentManager fragmentManager) {
         FragmentTransaction ft = fragmentManager.beginTransaction();
-        Fragment hangoutHistoryFragment = new HangoutsFragment().newInstance(user);
+        ArrayList<String> conditions = new ArrayList<>(Arrays.asList("past", "user"));
+        Fragment hangoutHistoryFragment = new HangoutsFragment().newInstance(conditions);
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/NavigationUtils.java:78]: Can inline this as well, like we did for CreateQuickMatchFragment

Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/NavigationUtils.java:78]: Why do we need to create this fragment with `.newInstance` but CreateQuickMatchFragment you can construct it directly?

File: app/src/main/java/com/example/friendly/adapters/HangoutsAdapter.java [hunk 5]
```diff
@@ -60,24 +61,31 @@ class ViewHolder extends RecyclerView.ViewHolder {
...
         private TextView tvHangoutDate;
+        private TextView tvHangoutLocation;
 
         public ViewHolder(@NonNull View itemView) {
             super(itemView);
             tvHangoutUser1 = itemView.findViewById(R.id.tvHangoutUser1);
             tvHangoutUser2 = itemView.findViewById(R.id.tvHangoutUser2);
             tvHangoutDate = itemView.findViewById(R.id.tvHangoutDate);
+            tvHangoutLocation = itemView.findViewById(R.id.tvHangoutLocation);
 //                currentUser = ParseUser.getCurrentUser();
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/adapters/HangoutsAdapter.java:72]: should remove this comment

File: app/src/main/java/com/example/friendly/adapters/HangoutsAdapter.java [hunk 6]
```diff
@@ -60,24 +61,31 @@ class ViewHolder extends RecyclerView.ViewHolder {
...
 
         public void bind(Hangout hangout) {
             // Bind the post data to the view elements
             tvHangoutUser1.setText(hangout.getUser1().getUsername());
-            if(hangout.getUser2() != null){
+            if (hangout.getUser2() != null) {
                 tvHangoutUser2.setText(hangout.getUser2().getUsername());
             }
-            tvHangoutDate.setText(hangout.getDate().toString());
+            String formattedDate = new SimpleDateFormat("MM/dd/yyyy, h a").format(hangout.getDate());
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/adapters/HangoutsAdapter.java:82]: Can create a constant for the date format string

File: app/src/main/java/com/example/friendly/fragments/HangoutsFragment.java [hunk 7]
```diff
@@ -78,9 +88,13 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
         adapter = new HangoutsAdapter(mContext, allHangouts);
         rvHangouts.setAdapter(adapter);
-        query.queryHangouts(adapter);
+//        query.queryHangouts(adapter, queryCondition);
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/fragments/HangoutsFragment.java:91]: should remove this

File: app/src/main/java/com/example/friendly/fragments/match/QuickMatchFragment.java [hunk 8]
```diff
@@ -30,5 +41,32 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        fm.beginTransaction();
+        ArrayList<String> conditions = new ArrayList<>(Arrays.asList("future", "quick"));
+        Fragment fragTwo = HangoutsFragment.newInstance(conditions);
+        ft.add(R.id.quickMatchHangouts, fragTwo);
+        ft.commit();
+
+        btnCreateQuickHangout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.i(TAG, "onClick LongTerm History button");
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/fragments/match/QuickMatchFragment.java:59]: Is this log intentionally left in?

**Review by NavinMahabir — COMMENTED** [review]
Great work! Please address my comments and let me re-review before merging

File: app/src/main/java/com/example/friendly/HangoutQuery.java (hunk 1 above)
Comment (carolyn-wang) [reply app/src/main/java/com/example/friendly/HangoutQuery.java:37]: Apply to all cases that match

File: app/src/main/java/com/example/friendly/NavigationUtils.java (hunk 4 above)
Comment (carolyn-wang) [reply app/src/main/java/com/example/friendly/NavigationUtils.java:78]: I was thinking I could CreateQuickMatchFragment directly since I don't need to pass any parameters into it. For HangoutsFragment(), I need to use .newInstance() to pass in the queryingConditions parameter.

Comment (NavinMahabir) [reply app/src/main/java/com/example/friendly/NavigationUtils.java:71]: I think it's better styling to consistently use .newInstance for creating fragments. @dhruhin, do you have input here?

File: app/src/main/java/com/example/friendly/activities/LoginActivity.java [hunk 9]
```diff
@@ -28,10 +28,6 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_login);
 
-        // TODO: remove
-//        NavigationUtils.goPreferencesActivity(LoginActivity.this);
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/activities/LoginActivity.java:32]: This should really just be removed instead of adding a todo for later

File: app/src/main/java/com/example/friendly/objects/Preference.java [hunk 10]
```diff
@@ -2,6 +2,7 @@
 
 import java.util.List;
 
+@Deprecated
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/objects/Preference.java:5]: Ideally deprecating this should be done in a separate PR. When do you plan to remove this file?

File: app/src/main/java/com/example/friendly/fragments/match/QuickMatchFragment.java [hunk 11]
```diff
@@ -30,5 +41,25 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        mActivity = getActivity();
+        btnCreateQuickHangout = view.findViewById(R.id.btnCreateQuickHangout);
+
+        FragmentManager fm = getParentFragmentManager();
+        FragmentTransaction ft = fm.beginTransaction();
+
+        fm.beginTransaction();
+        ArrayList<String> conditions = new ArrayList<>(Arrays.asList("future", "quick"));
+        Fragment fragTwo = HangoutsFragment.newInstance(conditions);
+        ft.add(R.id.quickMatchHangouts, fragTwo);
```
Comment (NavinMahabir) [inline app/src/main/java/com/example/friendly/fragments/match/QuickMatchFragment.java:53]: Don't really need `fragTwo` since it's only used once. Also why did you name it `fragTwo` instead of something like `newHangoutsFragment`?

**Review by NavinMahabir — APPROVED** [review]
Approved to merge, but some small comments to review before doing so
